packages/modalConnector/src/ModalConnector.ts

```diff
@@ -0,0 +1,162 @@
+import { ethers } from 'ethers'
+import {
+  configureChains,
+  createClient,
+  getProvider,
+  getAccount,
+  getNetwork,
+  fetchSigner,
+  disconnect,
+  switchNetwork,
+  mainnet,
+  watchAccount,
+  watchNetwork,
+  GetAccountResult,
+  GetNetworkResult,
+  Chain,
+  watchProvider,
+  GetProviderResult
+} from '@wagmi/core'
+import { AbstractWeb3Connector } from '@ripio/sdk/connectors'
+import { Web3Modal } from '@web3modal/html'
+import {
+  EthereumClient,
+  modalConnectors,
+  walletConnectProvider
+} from '@web3modal/ethereum'
+import { errors } from '@ripio/sdk/types'
+
+export default class ModalConnector extends AbstractWeb3Connector {
+  web3Modal: Web3Modal
+  private _requestAccount: boolean
+  private _events: Array<CallableFunction> = []
+
+  constructor(
+    projectId: string,
+    chains: Chain[] | Chain = [mainnet],
+    requestAccount = true
+  ) {
+    super()
+    this._requestAccount = requestAccount
+
+    const _chains: Chain[] = Array.isArray(chains) ? chains : [chains]
+    // Wagmi Core Client
+    const { provider } = configureChains(_chains, [
+      walletConnectProvider({ projectId })
+    ])
+    const wagmiClient = createClient({
+      autoConnect: true,
+      connectors: modalConnectors({
+        projectId,
+        version: '1', // or "2"
+        appName: 'web3Modal',
+        chains: _chains
+      }),
+      provider
+    })
+
+    // Web3Modal and Ethereum Client
+    const ethereumClient = new EthereumClient(wagmiClient, _chains)
+    this.web3Modal = new Web3Modal({ projectId }, ethereumClient)
+  }
+
+  /**
+   * It activates the provider, sets the chainId, requests an account, subscribes to events, and
+   * detects the legacy chain
+   * @returns {object} provider, chainId and account
+   */
+  async activate(): Promise<{
+    provider: ethers.providers.StaticJsonRpcProvider
+    chainId: number
+    account: ethers.providers.JsonRpcSigner | ethers.Wallet | undefined
+  }> {
+    this._provider = getProvider<ethers.providers.StaticJsonRpcProvider>()
+    const { chain } = getNetwork()
+    this._chainId = chain?.id
+    if (this._requestAccount) {
+      // if _requestAccount was left as true (default value), request account, else, leave account null.
+      await this.requestAccount()
+    }
+    this.subscribeToEvents()
+    this._isActive = true
+    await this.detectLegacyChain()
+
+    return {
+      provider: <ethers.providers.StaticJsonRpcProvider>this._provider,
+      chainId: <number>this.chainId,
+      account: <ethers.providers.JsonRpcSigner>this._account
+    }
+  }
+
+  async deactivate(): Promise<void> {
+    this.unsubscribeFromEvents()
+    await super.deactivate()
+    await disconnect()
+  }
+
+  subscribeToEvents() {
+    this._events.push(
+      watchAccount(this.handleAccountChanged),
+      watchNetwork(this.handleChainChanged),
+      watchProvider({}, this.handleProviderChanged)
+    )
+  }
+
+  protected handleAccountChanged = async (account: GetAccountResult) => {
```

Comment:
Since we are implementing the same functions than in the BrowserWeb3Connector, we may think about extract them to the upper class as abstract methods that must be implemented, and in case that are not implemente/overwritten, just raise a not implemented exception if they are called. Same goes for the following handleX functions